### ISSUE TITLE
Ml mini project

### ISSUE THREAD

**Issue by niranja1533-arch** [niranja1533-arch/Ml-mini-project-#1]
import pandas as pd

import numpy as np

import matplotlib.pyplot as plt

from sklearn.linear_model import LinearRegression

from sklearn.metrics import mean_absolute_error, mean_squared_error

# Load dataset

data = pd.read_csv("ML\Forecast Revenue\subscription_data.csv")

# Convert date column to datetime

data['Order Date'] = pd.to_datetime(data['Order Date'])

# Sort data by date

data = data.sort_values('Order Date')

# Extract month and year

data['Year'] = data['Order Date'].dt.year

data['Month'] = data['Order Date'].dt.month

# Aggregate monthly revenue

monthly_data = data.groupby(['Year', 'Month'])['Sales'].sum().reset_index()

print(monthly_data.head())

# Create time index (important for regression)

monthly_data['Time_Index'] = np.arange(len(monthly_data))

# Features and target

X = monthly_data[['Time_Index']]

y = monthly_data['Sales']

# Train model

model = LinearRegression()

model.fit(X, y)

# Predictions on existing data

y_pred = model.predict(X)

# Forecast future months (next 6 months)

future_index = np.arange(len(monthly_data), len(monthly_data) + 6).reshape(-1, 1)

future_pred = model.predict(future_index)

# Evaluation

mae = mean_absolute_error(y, y_pred)

rmse = np.sqrt(mean_squared_error(y, y_pred))

print("MAE:", mae)

print("RMSE:", rmse)

# Plot actual vs predicted

plt.figure(figsize=(8,5))

plt.plot(monthly_data['Time_Index'], y, label='Actual Revenue')

plt.plot(monthly_data['Time_Index'], y_pred, label='Predicted Revenue')

plt.xlabel("Time (Months)")

plt.ylabel("Revenue")

plt.title("Monthly Revenue Forecast")

plt.legend()

plt.show()

# Plot future forecast

plt.figure(figsize=(8,5))

plt.plot(monthly_data['Time_Index'], y, label='Actual Revenue')

plt.plot(future_index, future_pred, label='Future Forecast', linestyle='dashed')

plt.xlabel("Time (Months)")

plt.ylabel("Revenue")

plt.title("Future Revenue Prediction (Next 6 Months)")

plt.legend()

plt.show()
